components/layout_thread/lib.rs

```diff
@@ -515,6 +521,7 @@ impl LayoutThread {
                 error_reporter: self.error_reporter.clone(),
                 local_context_creation_data: Mutex::new(local_style_context_creation_data),
                 timer: self.timer.clone(),
+                quirks_mode: self.quirks_mode.unwrap_or(QuirksMode::NoQuirks),
```

Comment:
I think it'd be wrong to lay out a quirks mode document in no-quirks mode, then laying it out in quirks mode, overall taking into account incremental reflows.